src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,495 @@
+    {
+        // We encode mCurrentSessions as a list of WebRTCSessionStruct::Type
+        return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
+            for (auto & session : mCurrentSessions)
+            {
+                ReturnErrorOnFailure(encoder.Encode(session));
+            }
+            return CHIP_NO_ERROR;
+        });
+    }
+
+    // If not our attribute, let default logic handle
+    return CHIP_NO_ERROR;
+}
+
+// CommandHandlerInterface
+void WebRTCTransportProviderServer::InvokeCommand(HandlerContext & ctx)
+{
+    ChipLogDetail(Zcl, "WebRTCTransportProvider: InvokeCommand called with CommandId=0x%08" PRIx32, ctx.mRequestPath.mCommandId);
+
+    if (mDelegate == nullptr)
+    {
+        ChipLogError(Zcl, "No delegate set for WebRTCTransportProviderServer");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::Failure);
+        return;
+    }
+
+    switch (ctx.mRequestPath.mCommandId)
+    {
+    case Commands::SolicitOffer::Id:
+        CommandHandlerInterface::HandleCommand<Commands::SolicitOffer::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleSolicitOffer(subCtx, req); });
+        break;
+
+    case Commands::ProvideOffer::Id:
+        CommandHandlerInterface::HandleCommand<Commands::ProvideOffer::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleProvideOffer(subCtx, req); });
+        break;
+
+    case Commands::ProvideAnswer::Id:
+        CommandHandlerInterface::HandleCommand<Commands::ProvideAnswer::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleProvideAnswer(subCtx, req); });
+        break;
+
+    case Commands::ProvideICECandidates::Id:
+        CommandHandlerInterface::HandleCommand<Commands::ProvideICECandidates::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleProvideICECandidates(subCtx, req); });
+        break;
+
+    case Commands::EndSession::Id:
+        CommandHandlerInterface::HandleCommand<Commands::EndSession::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleEndSession(subCtx, req); });
+        break;
+
+    default:
+        // Mark unrecognized command as UnsupportedCommand
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::UnsupportedCommand);
+        break;
+    }
+}
+
+// Helper functions
+WebRTCSessionStruct * WebRTCTransportProviderServer::FindSession(uint16_t sessionId)
+{
+    for (auto & session : mCurrentSessions)
+    {
+        if (session.id == sessionId)
+        {
+            return &session;
+        }
+    }
+
+    return nullptr;
+}
+
+void WebRTCTransportProviderServer::AddOrUpdateSession(const WebRTCSessionStruct & session)
+{
+    bool found = false;
+
+    // Search for a session with the same ID and update if found
+    for (auto & existing : mCurrentSessions)
```

Comment:
if we use stl anyway, would it make sense to have some std::find_if or std::find here?